### ISSUE TITLE
Add shortcuts to the editor page

### ISSUE THREAD

**Issue by greenlaw110** [freewind/rythmfiddle#42]
- when renaming or adding new template file, press Escape should cancel the process
- when it is inside the save fiddle dialog, press enter should trigger save and close the dialog
- press ctrl-s, popup save fiddle dialog
- press ctrl-shift-n, trigger create new fiddle
- press ctrl-n, trigger add new template
- press ctrl-alt-t, move focus to template editor
- press ctrl-alt-p, move focus to param editor

Optional:
- press alt-r, trigger click event on @Rythm in the top bar
- press alt-a, trigger click event on "All Demos" in the top bar
- press ctrl-i, trigger import demo click event

**Comment by freewind** [comment]
After trying [jquery.hotkeys](https://github.com/jeresig/jquery.hotkeys) for a while, I have to say it's not an easy task to add these shortkeys.

I tried to add bind `ctrl+s` on `$window.document`, it may popup the save dialog the first time, but when I close it, it won't be popped up again. 

And I tried to bind `ctrl+shift+n`, which is just not working.

The following code is inside `EditorCtrl` in `controllers.js`.

```
    $($window.document).bind('keydown', 'ctrl+s', function (evt) {
            $scope.save();
            return false;
    });
```

And there is also a big problem: The code mirror has its own key event handler. When I click in a codemirror, and press `ctrl+s`, the `template save dialog` won't show, instead, the system save dialog shown. So I have to add all these keyshorts to codemirror again.

Angularjs is not fit to "editor-like" application, I think we have to spend a lot of time make this feature work well.
